tempeh/datasets/seaphe_datasets.py

```diff
@@ -100,11 +124,23 @@ def __init__(self, drop_race=True):
             self._race_train = recover_categorical_encoding_for_compas_race(self._X_train)
             self._race_test = recover_categorical_encoding_for_compas_race(self._X_test)
 
-            # race is in columns 0-1 because the super class constructor removes the target
+            # race is in columns 0-1
             self._X_train = np.delete(self._X_train, np.s_[0:2], axis=1)
             self._X_test = np.delete(self._X_test, np.s_[0:2], axis=1)
             del[self._features[0:2]]
 
+        if drop_gender:
+            starting_column = 0 if drop_race else 2
+            self._gender_train = \
+                recover_categorical_encoding_for_compas_gender(self._X_train, starting_column)
```

Comment:
the handling of removal of some of the columns seems quite convoluted. we shouldn't need to deal with "starting_column" and/or column names since data is represented as a dataframe. also it is quite confusing to have a method whose name includes "compas". can we make things more transparent by dropping recover_* methods and creating/removing columns directly based on their names here (and for race too)?

Reply:
There are quite a few things to address here:
1. drop_<sensitive feature> is meant to provide flexibility. Not everyone wants to get rid of them, so with a simple flag we can allow dropping or keeping the features. That's a good thing.
2. starting_column just tells us where to drop the next attribute, which is a technicality. The comment states what it's for. Users don't need to know the details of what's happening under the hood, so I don't mind this little hack for now. More on the general idea at the end... 
3. The "recover" methods are there so that we can have numerical features (nominal (incl. binary) or continuous) but still string based sensitive features since sensitive features don't necessarily need to be part of X itself. This looks as hacky as it does because we store things as numpy arrays right now. More on that below...

Overall, we need to move to something more generic. The representation with numpy arrays isn't as flexible as we'd like, but we're kind of stuck with this for now because certain users are using it that way. If this gets funding I'm happy to address all these points, otherwise we'll have to live with them a little longer. This PR is just meant to get things going in the fairlearn repo, and for that it's good enough (and not exposed).